src/Api.Rest.Dtos.Tests/Compatibility/CompatibilityTests.cs

```diff
@@ -0,0 +1,404 @@
+
+			new CatalogAttributeDefinitionDto { Key = 13, Description = "Test", QueryEfficient = false, Catalog = new Guid( "11D4115C-41A7-4D47-A353-AF5DF61503EA" ) },
+
+			new AttributeDto( Characteristic.MeasurementModule, "Abweichungen" ),
+			new AttributeDto( Characteristic.X, 2919.0 ),
+			new AttributeDto( Characteristic.AlignmentType, 0 ),
+			new AttributeDto( Measurement.Time, new DateTime( 2010, 11, 04, 19, 44, 52, 8, DateTimeKind.Utc ) ),
+			new AttributeDto( Measurement.Time, (object)null ),
+
+			new CatalogDto
+			{
+				Name = "TestCatalog",
+				Uuid = new Guid( "8B23C6BB-EE2A-4BB3-A627-512C911F6FE8" ),
+				ValidAttributes = new[] { Catalog.OperatorNumber, Catalog.OperatorName },
+				CatalogEntries = new[]
+				{
+					new CatalogEntryDto
+					{
+						Key = 1,
+						Attributes = new[]
+						{
+							new AttributeDto( Catalog.OperatorNumber, 13 ),
+							new AttributeDto( Catalog.OperatorName, "Müller" ),
+						}
+					},
+					new CatalogEntryDto
+					{
+						Key = 2,
+						Attributes = new[]
+						{
+							new AttributeDto( Catalog.OperatorNumber, 42 ),
+							new AttributeDto( Catalog.OperatorName, "Meier" ),
+						}
+					},
+					new CatalogEntryDto
+					{
+						Key = 3,
+						Attributes = new[]
+						{
+							new AttributeDto( Catalog.OperatorNumber, 73 ),
+							new AttributeDto( Catalog.OperatorName, "Schulze" ),
+						}
+					}
+				}
+			},
+
+			new ConfigurationDto
+			{
+				PartAttributes = new[]
+				{
+					new AttributeDefinitionDto { Key = Part.Number, Description = "Teilenummer", Length = 30, Type = AttributeTypeDto.AlphaNumeric },
+					new AttributeDefinitionDto { Key = Part.VariantOfLine, Description = "Modell/Variante", Length = 40, Type = AttributeTypeDto.AlphaNumeric }
+				},
+
+				CharacteristicAttributes = new AbstractAttributeDefinitionDto[]
+				{
+					new AttributeDefinitionDto { Key = Characteristic.Number, Description = "Merkmalsnummer", Length = 20, Type = AttributeTypeDto.AlphaNumeric },
+					new AttributeDefinitionDto { Key = Characteristic.ControlItem, Description = "Dokumentationspflicht", Type = AttributeTypeDto.Integer },
+					new CatalogAttributeDefinitionDto { Key = 2004, Description = "Richtung", Catalog = new Guid( "d7291afb-0a67-4c1e-8bcc-6fc455bcc0e5" ) },
+				},
+
+				MeasurementAttributes = new[]
+				{
+					new AttributeDefinitionDto { Key = Measurement.Time, Description = "Zeit", Type = AttributeTypeDto.DateTime },
+					new AttributeDefinitionDto { Key = 1, Description = "Messwert", Type = AttributeTypeDto.Float }
+				},
+
+				ValueAttributes = new AbstractAttributeDefinitionDto[]
+				{
+				},
+
+				CatalogAttributes = new[]
+				{
+					new AttributeDefinitionDto { Key = 2009, Description = "Richtung", Length = 10, Type = AttributeTypeDto.AlphaNumeric }
+				},
+
+				VersioningType = VersioningTypeDto.Off
+			},
+
+			new DataMeasurementDto

```

Comment:
For completeness you can add some measured value attributes when defining the DataValueDto, e.g.
```csharp
Value = new DataValueDto
{
	Attributes = new[]
	{
		new AttributeDto( Value.MeasuredValue, 0.15108030390438515 ),
		new AttributeDto( Value.AggregatedCp, 0.25108030390438873 ),
		new AttributeDto( Value.AggregatedMaximum, 0.55108030390438542 )
	}
}
```

You can also add a test case for the SimpleMeasurementDto. Other than that thumbs up for including almost everything, even part history!